src/upgrade/nc_upgrade_manager.js

```diff
@@ -176,20 +176,22 @@ class NCUpgradeManager {
         if (!err_message && !hostnames.length) {
             err_message = `config dir upgrade can not be started missing hosts_data hosts_data=${util.inspect(hosts_data)}`;
         }
-        const missing_expected_hosts = !(expected_hosts.every(item => hostnames.includes(item)));
-        const missing_hostnames = !(hostnames.every(item => expected_hosts.includes(item)));
 
-        if (!err_message && missing_expected_hosts) {
-            err_message = `config dir upgrade can not be started - expected_hosts missing one or more hosts specified in system.json  hosts_data=${util.inspect(hosts_data)}`;
+        const all_hostnames_exist_in_expected_hosts = hostnames.every(item => expected_hosts.includes(item));
+        if (!err_message && !all_hostnames_exist_in_expected_hosts) {
```

Comment:
If it is just for adding the WARN in the logs, why do you still print it only if you don't have an `err_message`? (It is not a suggestion, adding it just to explain what I meant - )
```suggestion
        if (!all_hostnames_exist_in_expected_hosts) {
```

BTW, I think you can update the JSDoc of this function, so we will have the cases explicitly written, as we behave differently (WARN if we didn't have a host in our config but was in the expected_hosts).

Reply:
@shirady I think we can remove it, it's old code